ngraph/test/backend/one_hot.in.cpp

```diff
@@ -23,235 +23,139 @@
-
-    // Create some tensors for input/output
-    auto a = backend->create_tensor(element::i32, shape_a);
-    copy_data(a, vector<int32_t>{2, 1, 0, 0, 3, 2, 1, 0});
-    auto result = backend->create_tensor(element::i32, shape_r);
-
-    auto handle = backend->compile(f);
-    handle->call_with_validate({result}, {a});
-    vector<int32_t> rv = read_vector<int32_t>(result);
-
-    EXPECT_EQ(rv[0], 0);
-    EXPECT_EQ(rv[1], 0);
-    EXPECT_EQ(rv[2], 1);
-
-    EXPECT_EQ(rv[3], 0);
-    EXPECT_EQ(rv[4], 1);
-    EXPECT_EQ(rv[5], 0);
-
-    EXPECT_EQ(rv[6], 1);
-    EXPECT_EQ(rv[7], 0);
-    EXPECT_EQ(rv[8], 0);
-
-    EXPECT_EQ(rv[9], 1);
-    EXPECT_EQ(rv[10], 0);
-    EXPECT_EQ(rv[11], 0);
-
-    // These are undefined since value is out of bounds
-    // EXPECT_EQ(rv[12], 0);
-    // EXPECT_EQ(rv[13], 0);
-    // EXPECT_EQ(rv[14], 0);
-
-    EXPECT_EQ(rv[15], 0);
-    EXPECT_EQ(rv[16], 0);
-    EXPECT_EQ(rv[17], 1);
-
-    EXPECT_EQ(rv[18], 0);
-    EXPECT_EQ(rv[19], 1);
-    EXPECT_EQ(rv[20], 0);
-
-    EXPECT_EQ(rv[21], 1);
-    EXPECT_EQ(rv[22], 0);
-    EXPECT_EQ(rv[23], 0);
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<int32_t>({2, 1, 0, 0, 3, 2, 1, 0});
+    test_case.add_expected_output<int32_t>(
+        shape_r, {0, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0});
+    test_case.run();
 }
 
 NGRAPH_TEST(${BACKEND_NAME}, one_hot_matrix_0)
 {
     Shape shape_a{3, 3};
-    auto A = make_shared<op::Parameter>(element::i32, shape_a);
     Shape shape_r{3, 3, 3};
-    auto r = make_shared<op::OneHot>(A, Shape{3, 3, 3}, 0);
+    auto A = make_shared<op::Parameter>(element::i32, shape_a);
+    auto r = make_shared<op::OneHot>(A, shape_r, 0);
     auto f = make_shared<Function>(r, ParameterVector{A});
 
-    auto backend = runtime::Backend::create("${BACKEND_NAME}");
-
-    // Create some tensors for input/output
-    auto a = backend->create_tensor(element::i32, shape_a);
-    copy_data(a,
-              vector<int32_t>{
-                  0, 1, 1, 2, 1, 0, 0, 2, 1,
-              });
-    auto result = backend->create_tensor(element::i32, shape_r);
-
-    auto handle = backend->compile(f);
-    handle->call_with_validate({result}, {a});
-    EXPECT_EQ((vector<int32_t>{1, 0, 0, 0, 0, 1, 1, 0, 0,
-
-                               0, 1, 1, 0, 1, 0, 0, 0, 1,
-
-                               0, 0, 0, 1, 0, 0, 0, 1, 0}),
-              read_vector<int32_t>(result));
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<int32_t>({0, 1, 1, 2, 1, 0, 0, 2, 1});
+    test_case.add_expected_output<int32_t>(
+        shape_r, {1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0});
+    test_case.run();
 }
 
 NGRAPH_TEST(${BACKEND_NAME}, one_hot_vector_many_categories)
 {
     // Imagenet has roughly 20,000 categories
-    uint32_t category_count = 20000;
+    constexpr uint32_t category_count = 20000;
```

Comment:
This might be confusing - this doesn't need to be a constexpr, it should rather just be const.

Reply:
My aim was to point it's a fixed value, const doesn't guarantee that. However, for compiler it shouldn't matter in this case.